### PR TITLE
.github: add initial workflow to run tests

### LINKED ISSUE #37: `c4test` segfaults with e.g. `basic1.c4m`
Running `c4test` sometimes works:

```text
$ build/c4test tests/basic1.c4m
info: Compiling from: tests/basic1.c4m
Processing module tests/basic1.c4m

[...]

Heap Usage: 27932 kb of 131072 kb (103139 kb free)
 Copied 7960 allocation records. Trashed 32380 records.
New Usage: 6304 kb (21627 kb collected)
```

and sometimes segfaults:

```console
$ build/c4test tests/basic1.c4m
info: Compiling from: tests/basic1.c4m
zsh: segmentation fault (core dumped)  build/c4test tests/basic1.c4m
```

It looks like this was introduced by 1e6a017302e4438034a300f5a21e0eb2aeb723c8 (the codegen PR, https://github.com/crashappsec/libcon4m/pull/31). The commit before that seems OK.

The segfaults still occur with https://github.com/crashappsec/libcon4m/pull/36.

### PR DESCRIPTION
**Pull request by ee7**
Add an initial workflow to run `dev run` in CI on the Linux and macOS runners. The output currently needs to be inspected manually, but this PR gives us a base to build on and could already help catch regressions like #37 (and make it easier to verify fixes on other platforms).

Note that currently:

- The `dev run` command exiting with an exit status of 0 does not imply that the tests pass.
- The workflow indicates success even when `dev run` segfaults.

We can improve such things over time.
 
Also use the latest stable GCC release (GCC 14.1) in the Linux job to help make newer compiler warnings more visible.

Closes: #34 (we can make follow-up tickets for e.g. the exit status)
Refs: #35
Refs: #37

---

To-do:

- [x] Fix error on macOS: `meson.build:170:14: ERROR: C shared or static library 'crypto' not found`
- [x] Resolve any conflict with `main` after we merge a PR that fixes #37

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        run: |
+          case "${RUNNER_OS}" in
+            Linux)
+              sudo apt install meson ninja-build
+              ;;
+            macOS)
+              brew install meson ninja
+              ;;
+            *)
+              exit 1
```
Comment (miki725) [inline .github/workflows/test.yml:60]: lets use github expressions for this:

```
- name: install deps (linus)
  if: runner.os == 'Linux'
  run: sudo apt install meson ninja build

- name: install deps (mac)
  if: runner.os == 'macOS'
  ...
```

**Review by miki725 — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (miki725) [reply .github/workflows/test.yml]: When split it just shows up better in GitHub logs. The command you actually care about is directly displayed vs being collapsed

**Review by miki725 — APPROVED** [review]

**Review by viega — COMMENTED** [review]
It doesn't seem to deal w/ the exit code / crashes.  For instance, it's certainly not working on Linux right now based on the runner output, but it still passes. 

Also, since I'm generally not working on a Linux box and switching is a bit of a PITA, would appreciate some insight there.

**Comment by miki725** [comment]
https://github.com/crashappsec/libcon4m/blob/833300442d95e6ca7b49104417873c0c671caa00/dev#L128-L139

looks like exit code is not honored in the `dev` script in general. not sure how that works on mac at all. we can either `set -e` in the script to bail out on any non-successful commands or explicitly handle the case when `./c4test` fails

**Comment by viega** [comment]
I don't understand the need to be rushing it in if it isn't where it needs to be. There's no need to overengineer the build; I don't want to do more than the bare minimum necessary (for instance with the dev script), because they're not a priority yet. Breaking this up feels like an invitation to bikeshed, and I would rather stay focused.

**Pull request merged**